pkg/storegateway/gateway.go

```diff
@@ -253,7 +253,7 @@ func (g *StoreGateway) starting(ctx context.Context) (err error) {
 	// At this point, if sharding is enabled, the instance is registered with some tokens
 	// and we can run the initial synchronization.
 	g.bucketSync.WithLabelValues(syncReasonInitial).Inc()
-	if err = g.stores.InitialSync(ctx); err != nil {
+	if err = services.StartAndAwaitRunning(ctx, g.stores); err != nil {
 		return errors.Wrap(err, "initial blocks synchronization")
```

Comment:
(nit) I'm not sure if we should aim fixing it in this PR, but the metrics and the comments around this one now looks off. I.e. this code doesn't "know" anything about "initial sync" any more. Maybe a `TODO` comment about the need to refactor it further later would make this chunk more clear.

Reply:
i think it's in scope of this PR because they weren't outdated before this PR. They're updated now